bridge-history-api/orm/l2_sent_msg.go

```diff
@@ -73,25 +73,53 @@ func (l *L2SentMsg) GetLatestSentMsgHeightOnL2(ctx context.Context) (uint64, err
 }
 
 // GetClaimableL2SentMsgByAddressWithOffset get claimable l2 sent msg by address with offset
-func (l *L2SentMsg) GetClaimableL2SentMsgByAddressWithOffset(ctx context.Context, address string, offset int, limit int) ([]*L2SentMsg, error) {
-	var results []*L2SentMsg
-	err := l.db.WithContext(ctx).Raw(`SELECT * FROM l2_sent_msg WHERE id NOT IN (SELECT l2_sent_msg.id FROM l2_sent_msg INNER JOIN relayed_msg ON l2_sent_msg.msg_hash = relayed_msg.msg_hash WHERE l2_sent_msg.deleted_at IS NULL AND relayed_msg.deleted_at IS NULL) AND (original_sender=$1 OR sender = $1) AND msg_proof !='' ORDER BY id DESC LIMIT $2 OFFSET $3;`, address, limit, offset).
-		Scan(&results).Error
-	if err != nil {
-		return nil, fmt.Errorf("L2SentMsg.GetClaimableL2SentMsgByAddressWithOffset error: %w", err)
+func (l *L2SentMsg) GetClaimableL2SentMsgByAddressWithOffset(ctx context.Context, address string, offset int, limit int) (uint64, []*L2SentMsg, error) {
+	var totalMsgs []*L2SentMsg
+	db := l.db.WithContext(ctx)
+	db = db.Table("l2_sent_msg")
+	db = db.Where("original_sender = ? OR sender = ?", address, address)
+	db = db.Where("msg_proof != ''")
+	db = db.Where("deleted_at IS NULL")
+	db = db.Order("id DESC")
+	if err := db.Find(&totalMsgs).Error; err != nil {
+		return 0, nil, err
+	}
+
+	msgHashes := make([]string, len(totalMsgs))
+	for i, msg := range totalMsgs {
+		msgHashes[i] = msg.MsgHash
+	}
+
+	var claimedMsgHahes []string
+	db = l.db.WithContext(ctx)
+	db = db.Table("relayed_msg")
+	db = db.Where("msg_hash IN ?", msgHashes)
```

Comment:
They say `IN` is quite slow with >100 values, and there are some alternatives like this https://postgres.cz/wiki/PostgreSQL_SQL_Tricks_I#Predicate_IN_optimalization but I'm not deeply knowledgeable about this.

Reply:
We should add a metric about these result set sizes in the future.

Reply:
added a comment about metric in [75bdd60](https://github.com/scroll-tech/scroll/pull/959/commits/75bdd606dca809d307de6dfe8cd2d13856b47273)
and changed query style to `VALUES(10),(20)`.

Reply:
Thanks. This syntax looks a bit unusual, can you test that it works before we merge this PR?

Reply:
It does not work (`db = db.Where(fmt.Sprintf("msg_hash IN (VALUES %s)", valuesStr))`), so I change it back.